UT4MasterServer/Controllers/AdminPanelController.cs

```diff
@@ -206,6 +223,108 @@ public async Task<IActionResult> DeleteTrustedServer(string id)
 		return Ok(ret);
 	}
 
+	[HttpPatch("change_password/{id}")]
+	public async Task<IActionResult> ChangePassword(string id, [FromBody] string newPassword, [FromBody] bool? iAmSure)
```

Comment:
Honestly just a plain `NotFound` sounds even better.

Reply:
Actually... All methods in `AdminPanelController` only allow admins to access it. `VerifyAdmin` method makes sure that's the case. Only users that are considered Admin or Moderator are able to interact with this controller. All other users will get 401 Unauthorized response anyway. All admins are able to see permissions of all users.

Are you saying that if an admin tries to change password of another admin they should only get BadRequest without any knowledge as to why?

Reply:
It is a good practice not to reveal any information that could be misused. In this case, if you have a malicious admin that is trying to get access to another's admin, you are giving him additional data (GUID of another user) that otherwise would be difficult to get. Generally, if it is same for functionality to return just BadRequest or NotFound, that is better solution.

Reply:
the id of any user is very trivial to get. i commonly looked up other's ids with `ut4stats.com`. I would consider it public knowledge, even player card in the game originally displayed it.

i am pretty sure that all admins will have knowledge about who are the other admins. If there's a malicious admin, then making that person an admin was a mistake in the first place. Admins are supposed to have access to everything.

I fail to understand in what way this response message can be misused.

If i am an admin and for example by mistake enter wrong id which is another admin and i get a response which just says "400 bad request", i'm going to be confused.

If i am a malicious admin and intentionally try to change other admin's account. no one will know. the only logical solution i see is to log seemingly suspicious activity.

Reply:
Never mind, if you say that is a publicly accessible information, hiding it won't help much.